src/set_relation/set_relation.cc

```diff
@@ -3574,7 +3581,153 @@ Relation* Relation::boundDomainRange()
     
     r->acceptVisitor(v);
 
+    delete v;
+
     return r;
 }
 
+/*****************************************************************************/
+#pragma mark -
+/*************** VisitorSuperAffineSet *****************************/
+// Vistor Class used in SuperAffineSet
+// Used in traversing a Set/Relation to replace UFCalls with symbolic constants
+// We will build up a affineSet (or affineRelation), term by term.
+// And if a term is UFCall, we will convert it to symbolic constant,
+// which is pre-computed and stored in the ufcmap
+class VisitorSuperAffineSet : public Visitor {
+  private:
+         UFCallMap* ufcmap;
+         Set* affineSet;
+         Relation* affineRelation;
+         Conjunction* affineConj;
+         bool rel;         // To know whether we are traversing a Relation
+  public:
+         VisitorSuperAffineSet(UFCallMap* imap){ ufcmap = imap;}
+         //! Builds up an affineExp and adds it to affineConj
+         void preVisitExp(iegenlib::Exp * e);
+         //! Initializes an affineConj
+         void preVisitConjunction(iegenlib::Conjunction * c);
+         //! adds the current affineConj to affineSet (affineRelation)
+         void postVisitConjunction(iegenlib::Conjunction * c);
+         //! Initializes affineSet and rel
+         void preVisitSet(iegenlib::Set * s);
+         //! Initializes affineRelation and rel
+         void preVisitRelation(iegenlib::Relation * r);
+
+         Set * getSet(){ return affineSet; }
+         Relation * getRelation(){ return affineRelation; }
+};
+
+void VisitorSuperAffineSet::preVisitExp(iegenlib::Exp * e)
+{
+    //! If this is a argument to a UFCall, we don't want to modify it. 
+    if( e->isExpression() ){
```

Comment:
Not sure this is true.  Do you have examples with nested uf calls like f(g(i), v(i))?